### PR TITLE
239 validate interval parts same length

### PR DESCRIPTION
**Pull request by RakeshBobba03**
This pull request implements two new operators `split_parts_have_equal_length` and `split_parts_have_not_equal_length` to address issue #239, which validates that interval date/time representations have the same completeness on both sides of the solidus. The operators split a value by a configurable separator (defaulting to "/") and verify that both resulting parts have equal string length, correctly validating examples like "2003-12-15T10:00/2003-12-15T10:30" while flagging violations like "2003-12-15T10:00/2003-12-15T10:30:15" where precision differs. The implementation is intentionally generic to support any paired string format requiring equal precision, handles all edge cases including nulls and empty strings, and includes unit tests for both PandasDataset and DaskDataset along with updated operator documentation.

### CODE REVIEW

File: cdisc_rules_engine/check_operators/dataframe_operators.py [hunk 1]
```diff
@@ -879,6 +879,62 @@ def shorter_than(self, other_value: dict):
...
+        results = split_series.apply(validate_split)
+
+        # Ensure nulls/empties are marked as valid (True)
+        results = results | is_null_or_empty
+
+        return results
+
+    @log_operator_execution
+    @type_operator(FIELD_DATAFRAME)
+    def split_parts_have_not_equal_length(self, other_value: dict):
```
Comment (gerrycampion) [inline cdisc_rules_engine/check_operators/dataframe_operators.py]: Can this be named `split_parts_have_unequal_length`?

File: tests/unit/test_check_operators/test_string_comparison.py [hunk 2]
```diff
@@ -703,3 +703,153 @@ def test_not_matches_regex(data, comparator, dataset_type, expected_result):
...
+    [
+        # SENDIG rule examples - valid interval date/times with equal precision
+        (
+            {
+                "target": [
+                    "2003-12-15T10:00/2003-12-15T10:30",
+                    "2003-12-01/2003-12-10",
+                    "2003-01-01/2003-06-30",
+                ]
+            },
```
Comment (gerrycampion) [inline tests/unit/test_check_operators/test_string_comparison.py]: instead of defining a dictionary here like `{"target": [1,2,3]}`, you could just define the inner array like `[1,2,3]` and build the target below like:
`df = dataset_type.from_dict({"target": data})`
It just simplifies the test cases a bit.

File: tests/unit/test_check_operators/test_string_comparison.py [hunk 3]
```diff
@@ -703,3 +703,153 @@ def test_not_matches_regex(data, comparator, dataset_type, expected_result):
         {"target": "target", "comparator": comparator}
     )
     assert result.equals(df.convert_to_series(expected_result))
+
+
+@pytest.mark.parametrize(
+    "data,separator,dataset_type,expected_result",
+    [
```
Comment (gerrycampion) [inline tests/unit/test_check_operators/test_string_comparison.py:710]: Since these first 4 test cases all have the same separator and dataset_type, why not combine into a single test case, combining the data, and expected result arrays?  Same for combining the last 2 cases into a single test case.

File: resources/schema/Operator.md
Comment (gerrycampion) [inline resources/schema/Operator.md:1]: You'll need to update `Operator.json` as well.

File: resources/schema/Operator.md [hunk 4]
```diff
@@ -391,6 +391,40 @@ Length comparison
 
 Complement of `has_equal_length`
 
+### split_parts_have_equal_length
+
+Splits a string by a separator and checks if both parts have equal length. Generic operator for validating paired data formats where both parts must have the same level of detail or precision.
+
+Parameters:
+- `separator`: The delimiter to split on (default: "/")
```
Comment (gerrycampion) [inline resources/schema/Operator.md:400]: You'll need to update rule editor check param for separator as well: https://github.com/cdisc-org/conformance-rules-editor/blob/main/docs/check_parameter.md

**Review by gerrycampion — CHANGES_REQUESTED** [review]
Some minor issues

File: resources/schema/Operator.json [hunk 5]
```diff
@@ -4,841 +4,521 @@
...
-      "required": ["operator", "suffix", "value"],
+      "required": ["operator"],
       "type": "object"
     },
     {
       "properties": {
-        "operator": {
-          "const": "suffix_is_contained_by"
-        }
+        "operator": { "const": "split_parts_have_unequal_length" }
```
Comment (gerrycampion) [inline resources/schema/Operator.json:447]: It looks like we are maintaining alphabetic order. These should go before "starts_with"
